src/explore-education-statistics-admin/src/pages/release/datablocks/components/chart/ChartMapCustomGroupsConfiguration.tsx

```diff
@@ -72,25 +72,51 @@ export default function ChartMapCustomGroupsConfiguration({
                 .test('noOverlap', 'Groups cannot overlap', function noOverlap(
                   value: number,
                 ) {
-                  return !(
-                    groups.length &&
-                    groups.some(group => {
-                      return value >= group.min && value <= group.max;
-                    })
+                  // show an error when:
+                  // - the min is in an existing group, or
+                  // - this new group overlaps with another group and the max isn't in a group
+                  // (if the max is in a group and the min isn't then we only want to show the
+                  // error on the max field as it's the one that requires action)
+                  /* eslint-disable react/no-this-in-sfc */
+                  const minIsInAGroup = groups.some(
+                    group => value >= group.min && value <= group.max,
                   );
+                  const maxIsInAGroup = groups.some(
+                    group =>
+                      this.parent.max >= group.min &&
+                      this.parent.max <= group.max,
+                  );
+                  const overlaps = groups.some(
+                    group => value <= group.max && group.min <= this.parent.max,
+                  );
+                  return !minIsInAGroup && !(overlaps && !maxIsInAGroup);
+                  /* eslint-enable react/no-this-in-sfc */
```

Comment:
Would rewrite this to use early returns so that we don't perform any more work than is necessary. The ternary is also a little hard to understand still because of all the negations.

One way of writing this might be something like:

```ts
// Some comment about checking the min
if (groups.some(group => value >= group.min && value <= group.max)) {
  return false;
}

// Some comment about overlapping another group
const overlaps = groups.some(
  group => value <= group.max && group.min <= this.parent.max,
);

if (overlaps) {
  // Some comment about why we're checking the max instead
  return !groups.some(
      group =>
        this.parent.max >= group.min &&
        this.parent.max <= group.max,
    );
}

return true;
```

Reply:
I've changed both to use early returns and have clearer messages now.